modules/integration/src/test/scala/scala/cli/integration/ScalafixTestDefinitions.scala

```diff
@@ -0,0 +1,304 @@
+            |  def a = {
+            |    val x = 1 // keep unused - exec only RedundantSyntax
+            |    "Foo"
+            |  }
+            |}
+            |""".stripMargin
+      }
+
+      expect(updatedContent == expected)
+
+    }
+  }
+
+  test("--scalafix-arg arg") {
+    val original: String =
+      """|package foo
+         |
+         |final object Hello { // keep `final` beucase of parameter finalObject=false
+         |  s"Foo"
+         |}
+         |""".stripMargin
+    val inputs: TestInputs = TestInputs(
+      os.rel / confFileName ->
+        s"""|rules = [
+            |  RedundantSyntax
+            |]
+            |""".stripMargin,
+      os.rel / "Hello.scala" -> original
+    )
+    val expectedContent: String = noCrLf {
+      """|package foo
+         |
+         |final object Hello { // keep `final` beucase of parameter finalObject=false
+         |  "Foo"
+         |}
+         |""".stripMargin
+    }
+
+    inputs.fromRoot { root =>
+      os.proc(
+        TestUtil.cli,
+        "scalafix",
+        ".",
+        "--scalafix-arg=--settings.RedundantSyntax.finalObject=false",
+        "--power",
+        scalaVersionArgs
+      ).call(cwd = root)
+      val updatedContent = noCrLf(os.read(root / "Hello.scala"))
+      expect(updatedContent == expectedContent)
+    }
+  }
+
+  test("--scalafix-conf arg") {
+    val original: String =
+      """|package foo
+         |
+         |final object Hello {
+         |  s"Foo"
+         |}
+         |""".stripMargin
+
+    val confFileName = "unusual-scalafix-filename"
+    val inputs: TestInputs = TestInputs(
+      os.rel / confFileName ->
+        s"""|rules = [
+            |  RedundantSyntax
+            |]
+            |""".stripMargin,
+      os.rel / "Hello.scala" -> original
+    )
+    val expectedContent: String = noCrLf {
+      """|package foo
+         |
+         |object Hello {
+         |  "Foo"
+         |}
+         |""".stripMargin
+    }
+
+    inputs.fromRoot { root =>
+      os.proc(
+        TestUtil.cli,
+        "scalafix",
+        ".",
+        s"--scalafix-conf=$confFileName",
+        "--power",
+        scalaVersionArgs
+      ).call(cwd = root)
+      val updatedContent = noCrLf(os.read(root / "Hello.scala"))
+      expect(updatedContent == expectedContent)
+    }
+  }
+
+  test("external rule") {
+    val original: String =
+      """|//> using scalafix.dep "io.github.ghostbuster91.scalafix-unified::unified:0.0.8"
+         |
+         |package foo
+         |
+         |object Hello {
+         |  val a = List[Int]()
+         |}
+         |""".stripMargin
+    val inputs: TestInputs = TestInputs(
+      os.rel / confFileName ->
+        s"""|rules = [
+            |  EmptyCollectionsUnified
+            |]
+            |""".stripMargin,
+      os.rel / "Hello.scala" -> original
+    )
+    val expectedContent: String = noCrLf {
+      """|//> using scalafix.dep "io.github.ghostbuster91.scalafix-unified::unified:0.0.8"
```

Comment:
> As scalafix is (usually) used across several files, I doubt that a directive is the best UI to inject community rules.

I think it's fine. In case of having multiply files you don't need to define this dep in every file, having it in a single file will be enough